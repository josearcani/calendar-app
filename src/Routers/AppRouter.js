import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { startChecking } from '../actions/authActions';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
  const { checking, uid } = useSelector((state) => state.auth);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);
  


  if (checking) {
    return <h1>Espere</h1>
  }

  return (
    <Router>
      <div>
        <Switch>

          <PublicRoute
            exact
            path="/login"
            component={ LoginScreen }
            isAuth={ !!uid }
          />
          <PrivateRoute
            exact
            path="/"
            component={ CalendarScreen }
            isAuth={ !!uid }
          />

          <Redirect to="/login" />
        </Switch>
      </div>
    </Router>
  )
};
