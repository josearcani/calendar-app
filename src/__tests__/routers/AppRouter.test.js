import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { AppRouter } from '../../Routers/AppRouter';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);




describe('Test on <AppRouter />', () => {
  test('should show loader', () => {
    const initState = {
      auth: {
        checking: true
      }
    };
    const store = mockStore(initState);
    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('h1').text().trim()).toBe('Espere');
  });
  
});
