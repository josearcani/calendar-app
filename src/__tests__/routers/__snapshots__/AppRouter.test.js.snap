// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> should show loader 1`] = `
<h1>
  Espere
</h1>
`;
